Migrate notices component CSS

blowery opened the pull request:
#### Changes proposed in this Pull Request

* Migrate notices, global-notices, people-notices CSS

#### Testing instructions

* Pull up notices in devdocs, they should still appear as on master

blowery committed:
Migrate notices

blowery committed:
Fix lint errors

blowery committed:
Migrate global notices

blowery committed:
Migrate people notices

jsnajdr commented on client/notices/index.js:
```diff
@@ -133,17 +157,19 @@ const notices = {
 
 	/**
-	 * Callback handler to clear notices when a user leaves current page
+	 * Callback middleware to clear notices when a user leaves current page
```
🙂 I actually prefer calling the page.js functions "handlers" (another popular name is "controller") and keep the "middleware" name for the Redux ones. Too many unrelated things are called middleware these days.

blowery replied:
fair enough :)

blowery replied:
```suggestion
	 * Callback handler to clear notices when a user leaves current page
```

jsnajdr commented on client/notices/index.js:
```diff
@@ -133,17 +157,19 @@ const notices = {
+			changed = false;
+		const isNoticePersistent = function( notice ) {
+			return notice.persistent;
+		};
```
While we're fixing the ESLint warnings here, a more modern ES6 way of using the `length` and `container` variables is to restrict them to the `for` loop's scope:
```js
for ( const container in list.containerNames ) {
  const { length } = list[ container ];
}
```
This wasn't possible before ES6, because the loop didn't have its own scope.

blowery replied:
done

jsnajdr reviewed (commented):
I actually tried to do this migration myself a few days ago, but then got stuck on this `@extend .global-notices` call:

https://github.com/Automattic/wp-calypso/blob/ead3819515e706347f2a721b313034f22809e48d/client/post-editor/editor-notice/style.scss#L29

It's in Classic Editor's post-publish preview, added in #15069 (that PR contains screenshots and explanation what's going on)

blowery commented:
> I actually tried to do this migration myself a few days ago, but then got stuck on this @extend .global-notices call:

What was the sticking point? Duplicating the styles from the `@import 'components/global-notices/style';`?

jsnajdr commented:
> What was the sticking point?

Nothing really serious -- one just needs to understand what the editor notices do and how to test them. Not the zero-effort task I expected 🙂

blowery committed:
Update client/notices/index.js

blowery committed:
modernize clearNoticesOnNavigation a bit

jsnajdr committed:
Remove the unused notices-list CSS class

jsnajdr committed:
Layout: remove unused forcePinned prop from GlobalNotices

jsnajdr committed:
Remove unused CSS rule for editor-notice

`.editor-notice` is never a child of `.post-editor__content-editor` today. Removing unused rule.

jsnajdr committed:
Avoid extending global-notices CSS in editor-notice

Instead, add the `global-notices` class to the editor notice element.
Extracted a `GlobalNoticesContainer` component to do that automatically.

jsnajdr commented:
I rebased this PR onto latest master and added a few more commits:

The `notices-list` CSS class and `forcePinned` prop are no longer used. The `GlobalNotices` component used to be more complex in the past.

The `EditorNotice` component that shows a notice in Classic Editor uses some rules from the `global-notices` class and does it by using `@extend .global-notices`:

<img width="504" alt="Screenshot 2019-03-13 at 16 07 21" src="https://user-images.githubusercontent.com/664258/54290683-b29f0180-45ab-11e9-90d9-3fc60c8bd5d6.png">

I removed the `extend` and added a `global-notices` to the `EditorNotice` instead, implementing the same behavior in a CSS-only way.

The `is-global` class name could also be eliminated, but I didn't do it in this PR.

**How to test:**
Verify that the `EditorNotice` is displayed with proper positioning on all screen sizes and that the element indeed has the right classes:
<img width="955" alt="Screenshot 2019-03-13 at 16 12 23" src="https://user-images.githubusercontent.com/664258/54290789-e2e6a000-45ab-11e9-95b7-e3c60bb1352e.png">

blowery commented:
@jsnajdr few test failures that need cleaning. nice work!

blowery committed:
fix up tests

jsnajdr reviewed (approved):
Tests are passing and manually tested people notices, editor notices and global notices look :+1:
